File: app/src/main/java/com/gooner10/popularmoviesapp/Activity/moviefavourite/MovieFavoritePresenter.java
package com.gooner10.popularmoviesapp.Activity.moviefavourite;

import android.content.Context;

import com.gooner10.popularmoviesapp.Activity.data.MovieRepositoryImpl;

/**
 * Presenter for the FavoriteFragment
 */

public class MovieFavoritePresenter implements MovieFavouriteContract.Presenter {
    private final MovieFavouriteContract.View view;
    private MovieRepositoryImpl movieRepository;
    private Context context;

    public MovieFavoritePresenter(Context context, MovieFavouriteContract.View view) {
        this.context = context;
        this.view = view;
    }

    @Override
    public void loadFavoriteMovieFromRepository() {
        movieRepository = MovieRepositoryImpl.getMovieDatabaseInstance(context);
        view.displayFavoriteMovie(movieRepository.getMovie());
    }
}

File: app/src/main/java/com/gooner10/popularmoviesapp/moviefavourite/MovieFavoritePresenter.kt
package com.gooner10.popularmoviesapp.moviefavourite

import android.content.Context

import com.gooner10.data.MovieRepositoryImpl

/**
 * Presenter for the FavoriteFragment
 */

class MovieFavoritePresenter(private val context: Context, private val view: MovieFavouriteContract.View) : MovieFavouriteContract.Presenter {
    private var movieRepository: com.gooner10.data.MovieRepositoryImpl? = null

    override fun loadFavoriteMovieFromRepository() {
        movieRepository = com.gooner10.data.MovieRepositoryImpl.getMovieDatabaseInstance(context)
        view.displayFavoriteMovie(movieRepository!!.movie)
    }
}
